### ISSUE TITLE
fixing an mda-buffer for an enveloppe stuck in the queue

### ISSUE THREAD

**Issue by ubitux** [OpenSMTPD/OpenSMTPD#797]
If you mistakenly set a failing command line for an mda (let's say `deliver as mda "fdm -wrong-option fetch"`), realize your mistake, and then decide to fix it after a message is stuck in the queue, there doesn't seem to be any way of getting them out because the envelopes keep the `mda-buffer` field containing the wrong command. Even after a restart or re-scheduling all the envelopes.

```
# smtpctl show queue
15bbfcb8f8b07f10|inet4|mda||foo@example.com|bar@example.com|bar@example.com|1500048642|1500394242|0|83|pending|1139|"usage: fdm [-hklmnqv] [-a name] [-D name=value] [-f conffile] [-u user] [-x name] [fetch|poll|cache] [arguments]"
```

```
# smtpctl show envelope 15bbfcb8f8b07f10 
version: 2
type: mda
smtpname: ...
helo: ...
hostname: ...
errorline: "usage: fdm [-hklmnqv] [-a name] [-D name=value] [-f conffile] [-u user] [-x name] [fetch|poll|cache] [arguments]"
sockaddr: ...
sender: foo@example.com
rcpt: bar@example.com
dest: bar@example.com
ctime: 1500048642
last-try: 0
last-bounce: 1500063042
expire: 345600
retry: 83
dsn-notify: 0
esc-class: 4
esc-code: 30
mda-buffer: fdm -wrong-option fetch
mda-method: mda
mda-user: ...
mda-usertable: <getpwnam>
```

Is it possible to fix the `mda-buffer` manually one way or another, or reload it from the configuration? (I suppose the later solution causes design issue to find back the command in the new configuration).

Edit: my current workaround was to stop `smtpd`, edit `/var/spool/smtpd/queue/15bbfcb8f8b07f10` to fix the `mda-buffer`, start again `smtpd`, and `schedule all`. If the same could be achieved without restarting (typically using `smtpctl`) that would be great.

**Comment by poolpOrg** [comment]
Currently there is no way to work around this, admin must edit the envelopes in queue and restart to have them reprocessed. There's a lengthy explanation why in the tickets, but there's ongoing work to make this work in the future: https://github.com/OpenSMTPD/OpenSMTPD/projects/2

**Comment by poolpOrg** [comment]
This will be naturally resolved with #772, since your ticket is now referenced there I'll close it.
